core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -678,11 +678,13 @@ class Partition(val topicPartition: TopicPartition,
                              isr: Set[Int],
                              addingReplicas: Seq[Int],
                              removingReplicas: Seq[Int]): Unit = {
-    remoteReplicasMap.clear()
+    val replicaSet = assignment.toSet
+    val removedReplicas = remoteReplicasMap.keys -- replicaSet
+
     assignment
       .filter(_ != localBrokerId)
       .foreach(id => remoteReplicasMap.getAndMaybePut(id, new Replica(id, topicPartition)))
-
+    removedReplicas.foreach(remoteReplicasMap.remove)
```

Comment:
I decided to not get fancy with refactorings - this is literally the old code (https://github.com/apache/kafka/blob/7f9187fe399f3f6b041ca302bede2b3e780491e7/core/src/main/scala/kafka/cluster/Partition.scala#L657))

Reply:
Would `remoteReplicasMap --= removedReplicas` work here?

Reply:
`remoteReplicasMap --= removedReplicas` doesn't compile - the `remoteReplicasMap` is using a Kafka `Pool` class which itself is using a Java Map and I don't think they support the `--=` notation

Reply:
Sounds good to introduce the method!